backend/layers/processing/process_validate_atac.py

```diff
@@ -0,0 +1,224 @@
+        :param dataset_version_id: the dataset version id
+        :param processing_status_key: the key to update the processing status
+        :param datasets_bucket: the bucket to upload the dataset to
+        :param fragment_artifact_id: the artifact id of the fragment file to be use in the fragment index file.
+        :return:
+        """
+        self.update_processing_status(dataset_version_id, processing_status_key, DatasetConversionStatus.UPLOADING)
+        try:
+            artifact_id = self.business_logic.add_dataset_artifact(dataset_version_id, artifact_type, "dummy")
+            if fragment_artifact_id:
+                key_prefix = self.get_key_prefix(fragment_artifact_id.id)
+            else:
+                key_prefix = self.get_key_prefix(artifact_id.id)
+            key = key_prefix + "." + ARTIFACT_TO_EXTENSION[artifact_type]
+            datasets_s3_uri = self.upload_artifact(file_name, key, datasets_bucket)
+            self.logger.info(f"Uploaded {dataset_version_id}.{artifact_type} to {datasets_s3_uri}")
+            self.business_logic.update_dataset_artifact(artifact_id, datasets_s3_uri)
+            self.logger.info(f"Updated database with {artifact_type}.")
+            self.update_processing_status(dataset_version_id, processing_status_key, DatasetConversionStatus.UPLOADED)
+            return artifact_id
+        except Exception as e:
+            self.logger.error(e)
+            raise ConversionFailed(processing_status_key) from None
+
+    def skip_atac_validation(
+        self, local_anndata_filename: str, manifest: IngestionManifest, dataset_version_id
+    ) -> bool:
+        """
+        Check if atac validation should be skipped
+        :param local_anndata_filename: the local anndata file
+        :param manifest: the manifest
+        :param dataset_version_id: the dataset version id
+        :return: True if the validation should be skipped, False otherwise
+        """
+        # check if the anndata should have a fragment file
+        try:
+            result = self.schema_validator.check_anndata_requires_fragment(local_anndata_filename)
+        except ValueError as e:  # fragment file forbidden
+            self.logger.warning(f"Anndata does not support atac fragment files for the follow reason: {e}")
+            if manifest.atac_fragment:
+                self.update_processing_status(
+                    dataset_version_id,
+                    DatasetStatusKey.VALIDATION,
+                    DatasetValidationStatus.INVALID,
+                )
+                raise ValidationAtacFailed(errors=[str(e), "Fragment file not allowed for non atac anndata."]) from None
+            self.logger.warning("Skipping fragment validation")
+            self.update_processing_status(
+                dataset_version_id,
+                DatasetStatusKey.ATAC_FRAGMENT,
+                DatasetConversionStatus.SKIPPED,
+                validation_errors=[str(e)],
+            )
```

Comment:
missing context for me--why do we need to set ATAC_STATUS as skipped during dataset initialization if we set it here?

Reply:
Also, advocating that if it truly doesn't requite fragment validation and one wasn't submitted, this should be NA rather than SKIPPED. since it was only skipped bc its not applicable to this dataset.

Reply:
The skipped is temporarily since ATAC is not part of the pipeline yet. It's removed [here](https://github.com/chanzuckerberg/single-cell-data-portal/pull/7458/files).

Reply:
got it, thanks

Reply:
in that case, def think this should be NA to distinguish from the SKIPPED case below